e2e/project_test.go

```diff
@@ -32,10 +46,124 @@ func TestProject(t *testing.T) {
 	assert.Equal(t, createRequest.Description, getResponse.Project.Description)
 	assert.ElementsMatch(t, createRequest.Labels, getResponse.Project.Labels)
 
+	// Delete Project completely (Created Labels stay)
 	_, err = ServerEndpoints.project.DeleteProject(context.Background(), &v1storageservices.DeleteProjectRequest{
 		Id: createResponse.GetId(),
 	})
 	if err != nil {
 		log.Fatalln(err.Error())
 	}
+
+	// Validating Project deletion by trying to get deleted Project which should fail and return nil
+	nilResponse, err := ServerEndpoints.project.GetProject(context.Background(), &v1storageservices.GetProjectRequest{
+		Id: getResponse.Project.Id,
+	})
+
+	assert.NotNil(t, err)
+	assert.Nil(t, nilResponse)
+}
+
+func TestProjectUsers(t *testing.T) {
+	// Create simple project with name and description
+	createRequest := &v1storageservices.CreateProjectRequest{
+		Name:        "Test Project 002",
+		Description: "This project is used to test that users cannot be added as duplicate.",
+	}
+
+	createResponse, err := ServerEndpoints.project.CreateProject(context.Background(), createRequest)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	projectID := uuid.MustParse(createResponse.Id)
+
+	// Add two individual users to Project
+	userId01 := uuid.New()
+	scope := []v1storagemodels.Right{v1storagemodels.Right(v1storagemodels.Right_RIGHT_READ)}
+	addUserResponse01, err := ServerEndpoints.project.AddUserToProject(
+		context.Background(),
+		&v1storageservices.AddUserToProjectRequest{
+			UserId:    userId01.String(),
+			Scope:     scope,
+			ProjectId: projectID.String(),
+		})
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	userId02 := uuid.New()
+	scope = []v1storagemodels.Right{v1storagemodels.Right(v1storagemodels.Right_RIGHT_WRITE)}
+	addUserResponse02, err := ServerEndpoints.project.AddUserToProject(
+		context.Background(),
+		&v1storageservices.AddUserToProjectRequest{
+			UserId:    userId02.String(),
+			Scope:     scope,
+			ProjectId: projectID.String(),
+		})
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	// Validate creation of Users
+	projectUsers, err := ServerEndpoints.project.ReadHandler.GetProjectUsers(projectID)
+	if err != nil {
+		log.Fatalln(err.Error())
+	}
+
+	assert.Equal(t, 3, len(projectUsers))
+
+	assert.NotNil(t, addUserResponse01)
+	assert.Equal(t, userId01, projectUsers[1].ID)
+	assert.Equal(t, projectID, projectUsers[1].ProjectID)
+
+	assert.NotNil(t, addUserResponse02)
+	assert.Equal(t, userId01, projectUsers[2].ID)
```

Comment:
Used the wrong ID to validate user creation. Don't know how I missed that.

```suggestion
	assert.Equal(t, userId02, projectUsers[2].ID)
```